src/cpp/fastdds/xtypes/utils.cpp

```diff
@@ -0,0 +1,1477 @@
+    TreeNodeType(
+            std::string member_name,
+            std::string type_kind_name,
+            DynamicType::_ref_type dynamic_type)
+        : member_name(member_name)
+        , type_kind_name(type_kind_name)
+        , dynamic_type(dynamic_type)
+    {
+    }
+
+    std::string member_name;
+    std::string type_kind_name;
+    DynamicType::_ref_type dynamic_type;
+};
+
+std::string type_kind_to_str(
+        const DynamicType::_ref_type& type);
+
+//// Implementation
+
+DynamicType::_ref_type container_internal_type(
+        const DynamicType::_ref_type& dyn_type)
+{
+    TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+    const auto ret = dyn_type->get_descriptor(type_descriptor);
+    if (ret != RETCODE_OK)
+    {
+        //throw utils::InconsistencyException("No Type Descriptor");
+    }
+    return type_descriptor->element_type();
+}
+
+std::vector<uint32_t> container_size(
+        const DynamicType::_ref_type& dyn_type)
+{
+    TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+    const auto ret = dyn_type->get_descriptor(type_descriptor);
+    if (ret != RETCODE_OK)
+    {
+        //throw utils::InconsistencyException("No Type Descriptor");
+    }
+    return type_descriptor->bound();
+}
+
+std::vector<std::pair<std::string, DynamicType::_ref_type>> get_members_sorted(
+        const DynamicType::_ref_type& dyn_type)
+{
+    std::vector<std::pair<std::string, DynamicType::_ref_type>> result;
+
+    std::map<MemberId, DynamicTypeMember::_ref_type> members;
+    dyn_type->get_all_members(members);
+
+    for (const auto& member : members)
+    {
+        ObjectName dyn_name = member.second->get_name();
+        MemberDescriptor::_ref_type member_descriptor {traits<MemberDescriptor>::make_shared()};
+        const auto ret = member.second->get_descriptor(member_descriptor);
+        if (ret != RETCODE_OK)
+        {
+            //throw utils::InconsistencyException("No Member Descriptor");
+        }
+        result.emplace_back(
+            std::make_pair<std::string, DynamicType::_ref_type>(
+                dyn_name.to_string(),
+                std::move(member_descriptor->type())));
+    }
+    return result;
+}
+
+std::string array_kind_to_str(
+        const DynamicType::_ref_type& dyn_type)
+{
+    auto internal_type = container_internal_type(dyn_type);
+    auto this_array_size = container_size(dyn_type);
+
+    std::stringstream ss;
+    ss << type_kind_to_str(internal_type);
+
+    for (const auto& bound : this_array_size)
+    {
+        ss << "[" << bound << "]";
+    }
+
+    return ss.str();
+}
+
+std::string sequence_kind_to_str(
+        const DynamicType::_ref_type& dyn_type)
+{
+    auto internal_type = container_internal_type(dyn_type);
+    auto this_sequence_size = container_size(dyn_type);
+
+    std::stringstream ss;
+    ss << "sequence<" << type_kind_to_str(internal_type);
+
+    for (const auto& bound : this_sequence_size)
```

Comment:
Review this logic.

Reply:
Compare bound with `static_cast<uint32_t>(LENGTH_UNLIMITED)` to cover the unbounded case.